### DISCUSSION TITLE
Add low-pass filter for "Muted" mod

### DISCUSSION THREAD

**Discussion by NotGumballer91** [ppy/osu#16740] [Ideas]
Currently, Lowering the volume is okay but muffling it sounds accurate and a nice touch too since that's exactly what it feels like before going deaf/muted.

**Comment by peppy** [comment]
Sounds like it might work well.

**Comment by frenzibyte** [comment]
I've went ahead and got a simple implementation for this that applies a high-pass filter to all relevant audio components (track, samples if enabled, and metronome if enabled):

https://user-images.githubusercontent.com/22781491/152198274-19239ac3-27d8-48fe-bccc-9fffd84280da.mp4

Still got a few things in consider, mainly about how harsh the filter is – thinking about combining it with a volume adjust and lowering down the cutoff (right now it tweens to the nyquist frequency, to completely mute the track)

**Reply by NotGumballer91** [reply]
Oh no. I think I made an misunderstanding here. 
I actually hoped it will be only **added in the menu** and not in the gameplay. 
Also, it should be a low-pass filter. Sorry for the mix-up.

**Reply by bdach** [reply]
> I actually hoped it will be only added in the menu and not in the gameplay.

that doesn't make any sense. mods are supposed to have the same effect on audio in menus they do during the actual gameplay.

**Reply by NotGumballer91** [reply]
About the cutoff, I think it should be like this:

![jeff](https://user-images.githubusercontent.com/64581009/152208619-7b7f1578-750e-417b-b563-f67506adaaf7.png)

Blue: volume
Red: audio filter
X-axis is the combo (default is 100)

Volume can be exponential as well though.

**Reply by hlysine** [reply]
I also played around with adding a low-pass filter, but in the end, it's barely noticeable. By the time the low-pass filter reaches a noticeable cutoff, the volume is already very low. Also, the low-pass filter essentially mutes the track by itself when it reaches 0 frequency, so perhaps we can just ditch the volume adjust and use the filter with a non-linear mapping between cutoff and combo?

**Reply by NotGumballer91** [reply]
> Also, the low-pass filter essentially mutes the track by itself when it reaches 0 frequency

That seems like a better idea, much simpler.

**Reply by frenzibyte** [reply]
> so perhaps we can just ditch the volume adjust and use the filter with a non-linear mapping between cutoff and combo?

Note that by doing that, the muted mod will have to lose its volume effect on song select, as applying such effects on the song select from a mod isn't possible yet 

Perhaps you could provide a local mixer owned by `MusicController` in the `ApplyToTrack` method specific for mods, which would be reset on `ResetTrackAdjustments`. That should definitely be PR'd separately, however.

**Reply by hlysine** [reply]
I'll take a look. Thanks for the hint.
